server/methods/core/shop.js

```diff
@@ -81,17 +80,24 @@ Meteor.methods({
     }
 
     // we'll accept a shop object, or clone the current shop
-    const seedShop = shopData || Collections.Shops.findOne(Reaction.getPrimaryShopId());
+    let seedShop = shopData;
+
+    if (_.isEmpty(seedShop)) {
+      const count = Collections.Shops.find().count() || "";
+      seedShop = Reaction.getPrimaryShop();
+      // generate unique shop name
+      seedShop.name = seedShop.name + count;
```

Comment:
Previously, `shopData.name` would get a number appended regardless of whether the shopData is a clone of PrimaryShop or passed by the calling function. Now, this only happens when Primary Shop is cloned.